### PR TITLE
Add cargo-unleash to CI

### LINKED ISSUE #150: Prepare for crates.io release
### Goal:

Make this repo is ready to be released on crates.io.

### Background:

We would like to setup CI to release crates in this repo to crates.io, but we will need to make sure we are ready before that.

Substrate is using [cargo-unleash](https://github.com/gnunicorn/cargo-unleash) and examples can be found here: https://github.com/paritytech/substrate/blob/master/.gitlab-ci.yml

### Actions:

Create a script that use cargo-unleash to detect any issues blocking release.

Put the script in `./scripts` folder.

Document all the findings here.

Fixes all the trivial problems.

Update the test Github Actions to run the unleash script to prevent regression.

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #150.

### CODE REVIEW

**Comment by KiChjang** [comment]
GitHub actions is properly gating on `cargo unleash check`, so this is a good sign.

**Comment by xlc** [comment]
You could update the nightly version used by Github Actions to whatever works.

And thanks for fixing all compile errors regards for Substrate alpha 7. Hopefully it will have proper semvar soon...

File: auction/src/lib.rs [hunk 1]
```diff
@@ -98,7 +98,10 @@ decl_error! {
 
 impl<T: Trait> Module<T> {
 	fn _on_finalize(now: T::BlockNumber) {
-		for (auction_id, _) in <AuctionEndTime<T>>::drain(&now) {
+		for (block_number, auction_id, _) in <AuctionEndTime<T>>::drain() {
```
Comment (xlc) [inline auction/src/lib.rs:101]: this isn't right, it should be using `drain_prefix`

https://github.com/paritytech/substrate/blob/b096036932c46a0b9da543fb826be20224a45bc7/frame/support/src/storage/mod.rs#L239

**Comment by KiChjang** [comment]
@xlc I'm trying to look at the test failures, nothing from the source code really jumped out at me as being wrong, could you take a look please?

**Comment by xlc** [comment]
@KiChjang will take a look now

**Comment by xlc** [comment]
Thanks for the work!

**Pull request merged**